process.go

```diff
@@ -682,5 +691,52 @@ func processImage(ctx context.Context) ([]byte, context.CancelFunc, error) {
 		checkTimeout(ctx)
 	}
 
+	if po.MaxBytes > 0 && canFitToBytes(po.Format) {
+		return processToFitBytes(ctx, img)
+	}
+
 	return img.Save(po.Format, po.Quality)
 }
+
+func processToFitBytes(ctx context.Context, img *vipsImage) ([]byte, context.CancelFunc, error) {
+	po := getProcessingOptions(ctx)
+
+	img.CopyMemory()
+	result, cancel, err := img.Save(po.Format, po.Quality)
+	if err != nil || len(result) < po.MaxBytes {
+		return result, cancel, err
+	}
+
+	initialResult := make([]byte, len(result))
+	copy(initialResult, result)
+	quality := po.Quality
+
+	for len(result) > po.MaxBytes {
+		cancel()
+
+		var diff float64
+		delta := float64(len(result)) / float64(po.MaxBytes)
+		switch {
+		case delta > 3:
+			diff = 0.25
+		case delta > 1.5:
+			diff = 0.5
+		default:
+			diff = 0.75
+		}
+		quality = int(float64(quality) * diff)
+		if quality < 10 {
+			return initialResult, func() {}, nil
+		}
+
+		checkTimeout(ctx)
+
+		img.CopyMemory()
```

Comment:
Don't need to copy memory before each save. It should be done once at the beginning of this func.